src/Lean/Linter/Util.lean

```diff
@@ -0,0 +1,39 @@
+import Lean.Data.Options
+import Lean.Elab.Command
+import Lean.Server.InfoUtils
+
+namespace Lean.Linter
+
+register_builtin_option linter.nolint : Bool := {
+  defValue := false,
+  descr := "disable all linters"
+}
+
+open Lean.Elab Lean.Elab.Command
+
+def publishMessage
+  (content : String) (range : String.Range) (severity : MessageSeverity := .information) : CommandElabM Unit :=
+do
+  let ctx := (← read)
+  let messages := (← get).messages |>.add (mkMessageCore ctx.fileName ctx.fileMap content severity range.start range.stop)
+  modify ({ · with messages := messages })
+
+abbrev SyntaxStack := List (Syntax × Nat)
+
+partial def anyWithStack (pred : Syntax → SyntaxStack → Bool) (stx : Syntax) : Bool :=
```

Comment:
If we assume that we will only need this to find nested `Syntax` by its lexical range, it might be easier to have something like
```
findSyntaxStack? (root child : Syntax) : Option SyntaxStack
```
where for performance we can skip recursing into nodes that do not contain `child.getRange`